packages/graphiql-plugin-explorer/vite.config.ts

```diff
@@ -2,21 +2,24 @@ import { defineConfig } from 'vite';
 import react from '@vitejs/plugin-react';
 import packageJSON from './package.json';
 
-// https://vitejs.dev/config/
+const IS_UMD = process.env.UMD === 'true';
+
 export default defineConfig({
   plugins: [react({ jsxRuntime: 'classic' })],
   build: {
+    // avoid clean cjs/es builds
+    emptyOutDir: !IS_UMD,
     lib: {
       entry: 'src/index.tsx',
       fileName: 'index',
       name: 'GraphiQLPluginExplorer',
-      formats: ['cjs', 'es', 'umd'],
+      formats: IS_UMD ? ['umd'] : ['cjs', 'es'],
     },
     rollupOptions: {
       external: [
         // Exclude peer dependencies and dependencies from bundle
         ...Object.keys(packageJSON.peerDependencies),
-        ...Object.keys(packageJSON.dependencies),
+        ...(IS_UMD ? [] : Object.keys(packageJSON.dependencies)),
```

Comment:
👏🏻 awesome!

this will make things nice and simple for esm/cjs, and the UMD bundle for CDN implementations